python/lsst/summit/utils/assessPeekExposure.py

```diff
@@ -0,0 +1,200 @@
+# This file is part of summit_utils.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+import atexit
+import logging
+import multiprocessing
+import shutil
+import time
+import warnings
+from pathlib import Path
+
+import lsst.afw.display as afwDisplay
+import matplotlib.pyplot as plt
+import numpy as np
+from astropy.table import Table
+from astropy.utils.exceptions import AstropyUserWarning
+from lsst.afw.geom import ellipses
+from lsst.daf.butler.datastore.cache_manager import DatastoreCacheManager
+from lsst.summit.utils.bestEffort import BestEffortIsr
+from lsst.summit.utils.peekExposure import PeekExposureTask
+from tqdm import tqdm
+
+# Set logger level to higher than CRITICAL to suppress all output
+silent_logger = logging.getLogger("silent_logger")
+silent_logger.setLevel(logging.CRITICAL + 1)
+
+
+# get one BestEffortIsr,afwDisplay per subprocess
+def initializer():
+    global bestEffort, pet, display, fig, ax
```

Comment:
Hope you've got a good reason for using globals here.

Might be a bit better to have one class to package up all the global state, just so there's only actual one global variable.

Reply:
Yeah, I admit this is a bit of a hack, but globals are the only way I know of to get variables out of the multiprocessing.Pool initializer function.  I don't think this script should ever be imported elsewhere so hopefully the hack is contained at least.  Wrapping into a class is a good idea.